screens/DoubleSearch.js

```diff
@@ -67,63 +88,85 @@ function DoubleSearch(props) {
+        else if (coordinatesFrom.longitude == coordinatesTo.longitude && coordinatesFrom.latitude == coordinatesTo.latitude) {
+            props.navigation.navigate("IndoorMapView", { From: from.name, To: to.name })
         }
-        else if (from.name == "Current Location" && !currentLocationCoords) {
-            return alert("Error: Are location services on?");
+        else if (coordinatesFrom && coordinatesTo) {
+            props.navigation.navigate("PreviewDirections", { From: coordinatesFrom, To: coordinatesTo, fromName:from.name, toName:to.name });
         }
         else {
-            return alert("The destination field is missing or you typed an invalid location. Please try again.");
+            return alert("The destination or origin field is missing or invalid. Please try again.");
         }
     };
 
     const destinationName = props.navigation.getParam("destinationName", "Destination");
+    const value = props.navigation.getParam("destinationIndex", "");
+    /**
+     * 
+     */
+    const fetchCurrentPosition = () => {
+        
+        getPosition().then(({coords}) => {
+            setCurrentLocationCoords({
+                    latitude: coords.latitude,
+                    longitude: coords.longitude
+                })
+            })
+            .catch((err) => {
+            alert (err.message);
+            });
+    }
 
     /**
-     * Algorithm to find the coordinates of a given building name.
+     * Algorithm to find the coordinates of a given building name or classroom name.
      * returns longitude and latitude
-     * TODO: B) Another algorithm or extend this one to take in consideration classrooms.
+     * 
      * A.U
      * @param {*} name 
      */
     const getCoordinates = (name) => {
 
-        var list = buildingData();
-        for (var key in list) {
-            if (list[key].name.includes(name)) {
-                return list[key].coordinates;
+        let buildingList = buildingData();
+        let classRoomsList = ClassRooms();
+        if (/\d/.test(name)) {
+            for (var key in classRoomsList) {
+                if(classRoomsList[key].room.includes(name)) {
+                const buildingCoords = buildingList[key].coordinates;
+                const isClassroom = {isClassRoom: name};
+                const result = {...buildingCoords, ...isClassroom};
+                return result;
+                }
             }
         }
-        if (name == "Current Location" && currentLocationCoords) {
-            getCurrentLocation();
-            setIsCurrentLocationFetched(true);
+        for (var key in buildingList) {
+            if (buildingList[key].name.includes(name)) {
+                return buildingList[key].coordinates;
+            } else if (buildingList[key].services.includes(name) || buildingList[key].departments.includes(name)) {
+                return buildingList[key].coordinates; 
+            }
```

Comment:
<img width="752" alt="Screen Shot 2020-04-05 at 5 47 29 PM" src="https://user-images.githubusercontent.com/35850933/78510735-88475580-7765-11ea-8ba4-828e15ad03af.png">

Reply:
Nice one! I can refactor the logic!